dashboard/client/src/pages/job/hook/useJobProgress.ts

```diff
@@ -16,30 +59,28 @@ export const useJobProgress = (jobId?: string) => {
   const [msg, setMsg] = useState("Loading progress...");
   const [error, setError] = useState(false);
   const [isRefreshing, setRefresh] = useState(true);
-  const refreshRef = useRef(isRefreshing);
   const onSwitchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setRefresh(event.target.checked);
   };
-  refreshRef.current = isRefreshing;
-  const { data: progress } = useSWR(
-    jobId ? ["useJobProgress", jobId] : null,
-    async (_, jobId) => {
-      const rsp = await getJobProgress(jobId);
-
-      setMsg(rsp.data.msg);
-      if (rsp.data.result) {
-        return rsp.data.data.detail;
-      } else {
-        setError(true);
-        setRefresh(false);
-      }
-    },
-    { refreshInterval: isRefreshing ? API_REFRESH_INTERVAL_MS : 0 },
+  const { data: tasks } = useFetchStateApiProgressByTaskName(
+    jobId,
+    isRefreshing,
+    setMsg,
+    setError,
+    setRefresh,
   );
 
+  const summed = (tasks ?? []).reduce((acc, task) => {
+    Object.entries(task.progress).forEach(([k, count]) => {
+      const key = k as keyof TaskProgress;
+      acc[key] = (acc[key] ?? 0) + count;
+    });
+    return acc;
+  }, {} as TaskProgress);
+
   const driverExists = !jobId ? false : true;
   return {
-    progress,
+    progress: summed,
```

Comment:
can you tell me what this syntax is called? Just to understand this part of code.

Reply:
javascript has a short-hand for defining dictionaries. If the key name and value variable has the same name, you can write only instead of the other.

So 
```
foo_value = 3
bar = 5

dict = {
  foo: foo_value,
  bar: bar
}
```
is the same as:
```
foo_value = 3
bar = 5

dict = {
  foo: foo_value,
  bar,
}
```